lib/SPIRV/SPIRVReader.cpp

```diff
@@ -1760,6 +1760,40 @@ Value *SPIRVToLLVM::transValueWithoutDecoration(SPIRVValue *BV, Function *F,
   case OpGetKernelNDrangeSubGroupCount:
     return mapValue(
         BV, transSGSizeQueryBI(static_cast<SPIRVInstruction *>(BV), BB));
+  case OpFPGARegINTEL: {
+    IRBuilder<> Builder(BB);
+
+    SPIRVFPGARegINTELInstBase *BC =
+        static_cast<SPIRVFPGARegINTELInstBase *>(BV);
+
+    Type *Int8PtrTyPrivate = Type::getInt8PtrTy(*Context, SPIRAS_Private);
+    IntegerType *Int32Ty = Type::getInt32Ty(*Context);
+
+    Value *UndefInt8Ptr = UndefValue::get(Int8PtrTyPrivate);
+    Value *UndefInt32 = UndefValue::get(Int32Ty);
+
+    StringRef AnnotStr("__builtin_intel_fpga_reg");
+    Constant *GS = BB->getModule()->getNamedGlobal("__reg");
+    if (!GS)
+      GS = Builder.CreateGlobalStringPtr(AnnotStr, "__reg");
+
+    Function *AnnotationFn;
```

Comment:
Please initialize by nullptr.

Reply:
I had reworked it in a resemblant way, but unfortunately, we still have to perform the cast of `Ty`. Because of this, going with the original flow ends up simpler.